Sources/Apollo/NetworkTransport.swift

```diff
@@ -5,28 +5,30 @@ public protocol NetworkTransport: class {
   
   /// Send a GraphQL operation to a server and return a response.
   ///
+  /// Note if you're implementing this yourself rather than using one of the batteries-included versions of `NetworkTransport` (which handle this for you): The `clientName` and `clientVersion` should be sent with any URL request which needs headers so your client can be identified by tools meant to see what client is using which request. The `addApolloClientHeaders` method is provided below to do this for you if you're using Apollo Graph Manager.
+  ///
   /// - Parameters:
   ///   - operation: The operation to send.
   ///   - completionHandler: A closure to call when a request completes. On `success` will contain the response received from the server. On `failure` will contain the error which occurred.
   /// - Returns: An object that can be used to cancel an in progress request.
   func send<Operation>(operation: Operation, completionHandler: @escaping (_ result: Result<GraphQLResponse<Operation>, Error>) -> Void) -> Cancellable
   
-  /// The name of the client to send as the `"apollographql-client-name"` header.
-  var clientName: String { get set }
+  /// The name of the client to send as a header value.
+  var clientName: String { get }
   
-  /// The version of the client to send as the `"apollographql-client-version"` header
-  var clientVersion: String { get set }
+  /// The version of the client to send as a header value.
+  var clientVersion: String { get }
 }
 
 public extension NetworkTransport {
   
-  /// The header field name for the Client Name
-  static var headerFieldNameClientName: String {
+  /// The field name for the Apollo Client Name header
+  var headerFieldNameApolloClientName: String {
```

Comment:
I have two questions about these header name properties:
- what was the reason that these changed to be instance variables, rather than static variables? (since `addApolloClientHeaders()` is in a generic context, I think it has access to `Self` to access the static properties, it certainly does with newer versions of Swift)
- at the moment these fields are not protocol requirements, so despite the fact that `addApolloClientHeaders()` accesses them through `self`, because that's happening in a generic context the `NetworkTransport` default implementation is used and any specialisation in an implementation is ignored, is that intentional? if that is intentional (and it would make sense to me that it is, because why would implementors want to customise the header field _names_?) it would be less surprising (at least to me) when reading the code if those were just static variables somewhere, not accessed through the protocol. 🙃

Reply:
So there's a couple thoughts happening here: 

- It seemed goofy to me that at the call site you had to specify the class name of the current class, although that's less bad with being able to call `Self` within the protocol (I thought the `Self` outside of the protocol thing hadn't landed yet?), and there was no real reason for them to be static rather than instance other than it seemed good for namespacing. However, now that by default they're mostly accessed within the protocol, I'll rethink this change.
- This is certainly in the realm of "my opinion" vs "the right way to do things", but I *hate* free-floating static variables. It's almost always primarily related to something, so I think it's way better to have static variables on whatever it is they're related to rather than floating in space.

Reply:
Cool, namespacing is good.
From reading the code I was unsure whether it was intentional that implementations of `NetworkTransport` were _not_ capable of specialising those properties because I saw them being accessed statically on implementation types:
https://github.com/apollographql/apollo-ios/search?q=headerFieldNameClientName&unscoped_q=headerFieldNameClientName

🙃